src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs

```diff
@@ -553,6 +644,7 @@ internal void ThrowIfRecordTypeHasValidationOnProperties()
         }
 
         modelType = Nullable.GetUnderlyingType(modelType) ?? modelType;
+        ParameterBindingMethodCache ??= new ParameterBindingMethodCache(throwOnInvalidMethod: false);
```

Comment:
Do we need to worry about multiple threads initializing this?

Reply:
Patched in https://github.com/dotnet/aspnetcore/pull/56827/commits/a29099b921877488e0338f1273a62dfa5f069730.